apps/parsley/src/context/LogContext/index.tsx

```diff
@@ -227,19 +229,21 @@ const LogContextProvider: React.FC<LogContextProviderProps> = ({
     });
 
     return results;
-  }, [
-    state.searchState.searchTerm,
-    processedLogLines,
-    upperRange,
-    lowerRange,
-    getLine,
-    dispatch,
-  ]);
+    // processedLogLines doesn't need to be in the dependency array because the search results would
+    // only change if a new filter is applied and a new filter cannot be applied if a search term
+    // is active since they use the same input. This is a perf optimzation that prevents search from
+    // being re-run when a section is toggled.
+  }, [state.searchState.searchTerm, upperRange, lowerRange, getLine, dispatch]); // eslint-disable-line react-hooks/exhaustive-deps
 
+  const openSectionAndScrollToLine = useOpenSectionAndScrollToLine(
+    processedLogLines,
+    sectioning.openSectionContainingLineNumber,
+    scrollToLine,
+  );
   const stringifiedSearchResults = searchResults.toString();
   useEffect(() => {
     if (searchResults.length > 0) {
-      scrollToLine(searchResults[0]);
+      openSectionAndScrollToLine(searchResults);
```

Comment:
`openSectionAndScrollToLine` needs all of the search results so it can open all of the sections that contain the search results.

Reply:
Thanks for explaining, now I understand why I was so confused by `useOpenSectionAndScrollToLine`. I think it's pretty confusing how `lineNumber` can be equal to an array of numbers (to open the sections) and also a single number (to scroll to). Besides the singular naming of `lineNumber`, it seems like to open the sections we could always expect an array of numbers and to scroll we could always expect a single number. Is it possible to separate that logic by using two separate variables or something otherwise?

Reply:
Opening sections won't always expect an array of numbers. These are the use cases:
-clicking on a bookmark opens 1 section and jumps to the bookmark line
-paginating search opens 1 section and jumps to the search result
-new search input opens all sections that contain the search result and jumps to the first search result. 
It's possible to separate the two, but the interface will become less clean in my opinion and the functions `UseSectionsResult.openSectionContainingLineNumber` is documented with JSDoc and an interface. 
Updating the interface will result in maintaining an object that looks like this: `{lineNumberToScrollTo: number, lineNumbersToOpen: number[]}` where the usage would look like this most the time:
```openSectionAndScrollToLine({lineNumbersToOpen: [searchResults[nextPage]], lineNumberToScrollTo:searchResults[nextPage]})````
instead of:
```openSectionAndScrollToLine(searchResults[nextPage])```